File: DesignPatternsDotNet/Decorator/OffsetPrinting.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Decorator
{
    public class OffsetPrinting : BookMaker {

        public OffsetPrinting(IBook book): base(book)
        {
            Console.WriteLine("Offset Printing");
	    }
        public string Description { get { return aBook.Description + ", offset printing"; } set { aBook.Description = value; } }
        public override double Price { get { return aBook.Price+200.00; } set { aBook.Price = value; } }
    }
}

File: DesignPatternsJava/src/main/java/com/decorator/OffsetPrinting.java
package com.decorator;
//Offset Printing costs 200 taka extra
public class OffsetPrinting extends BookMaker {
	public OffsetPrinting(IBook book) {
		super(book);
		System.out.println("Offset Printing");
	}
	public String getDescription() {
		return aBook.getDescription()+", offset printing";
	}
	public double getPrice() {
		return aBook.getPrice()+200.00;
	}
}
